pkg/controller/replicaset/replica_set.go

```diff
@@ -209,10 +215,32 @@ func (rsc *ReplicaSetController) getPodReplicaSet(pod *api.Pod) *extensions.Repl
 		sort.Sort(overlappingReplicaSets(rss))
 	}
 
-	// update lookup cache
-	rsc.lookupCache.Update(pod, rss[0])
+	ret_rs = rss[0]
 
-	return rss[0]
+	// check ownerReference.Controller and UID to determine if the rc controls the pod
+	// in case of multiple controllers match the pod's label
+	if controllerRef != nil {
+		ret_rs = nil
+		for _, rs := range rss {
+			if controllerRef.UID == rs.ObjectMeta.UID {
+				ret_rs = rs
+				break
+			}
+		}
+	}
+
+	if ret_rs == rss[0] {
+		// update lookup cache only for the oldest controller
```

Comment:
So this is according to the main rule by which handling of multiple controllers is done, i.e. for every pod with the same namespase&label to return the oldest one.